### ISSUE TITLE
Implemented tests for org.eclipse.smarthome.config.dispatch.internal.ConfigDispatcher

### ISSUE THREAD

**Issue by petarnv** [eclipse/smarthome#2106]
I have implemented several tests for org.eclipse.smarthome.config.dispatch.internal.ConfigDispatcher.
They test mainly different cases for processing files from configuration directory and its subdirectory.

I have found some unexpected behavior of the ConfigDispatcher:
    1) **Incorrect Overriding** - The last modified file in the configuration directory is not guaranteed to be the last processed file, which may cause an incorrect overriding of some of the properties.
    2) **No Update On Delete** - The configuration is not updated if a property is deleted from a file.

I will create a PR later.

**Comment by triller-telekom** [comment]
@petarnv I do see that we have a `ConfigDispatchTest.groovy`, but I do not see the tests related to this issue? Does the issue still exist? If so, how did you reproduce it?

**Comment by petarnv** [comment]
@triller-telekom, here is a link to the merged PR for the tests - https://github.com/eclipse/smarthome/pull/2108. There is a discussion about both of the above issues in that PR. 

The first one (Incorrect Overriding) is fixed and there are couple of tests for it:
https://github.com/eclipse/smarthome/pull/2108/files#diff-3b462c420d8c1c8102ccca1c6c67da42R462
https://github.com/eclipse/smarthome/pull/2108/files#diff-3b462c420d8c1c8102ccca1c6c67da42R512
https://github.com/eclipse/smarthome/pull/2108/files#diff-3b462c420d8c1c8102ccca1c6c67da42R557
https://github.com/eclipse/smarthome/pull/2108/files#diff-3b462c420d8c1c8102ccca1c6c67da42R588

The second one (No Update On Delete) turns out not to be an issue, but an expected behavior. The tests for it were verifying a functionality that wasn't there, so they were removed.

So, basically, I don't think this issue still exists and we can close it.

**Comment by triller-telekom** [comment]
@petarnv Thank you for the clarification.
